packages/web3-utils/src/socket_provider.ts

```diff
@@ -369,7 +371,11 @@ export abstract class SocketProvider<
 
 	protected _onMessage(event: MessageEvent): void {
 		const responses = this._parseResponses(event);
-		if (!responses) {
+		if (responses.length === 0) {
+			// no responses means lost connection, autoreconnect if possible
+			if (this._reconnectOptions.autoReconnect) {
+				this._reconnect();
```

Comment:
Is it possible to not decrease coverage of unit tests ( for newly added code lines ) thanks Alex